overlay/agent.cpp

```diff
@@ -933,14 +933,25 @@ Future<Nothing> ManagerProcess::__configureDockerNetwork(
   // However, Docker disallows this. So we will install a de-funct
   // rule in the DOCKER-ISOLATION chain to bypass any isolation
   // docker might be trying to enforce.
-  const string iptablesCommand = "iptables -w -D DOCKER-ISOLATION -j RETURN; "
-    "iptables -w -I DOCKER-ISOLATION 1 -j RETURN";
+  std::string iptablesCommand = 
+      "iptables -w -D DOCKER-ISOLATION -j RETURN; "
+      "iptables -w -I DOCKER-ISOLATION 1 -j RETURN";
 
   Duration timeout = Milliseconds(networkConfig.command_timeout());
-  return runScriptCommand(iptablesCommand, timeout)
-    .then([]() -> Future<Nothing> {
-        return Nothing();
-        });
+  Future<std::string> response = runScriptCommand(iptablesCommand, timeout);
+
+  response.await();
```

Comment:
This is blocking worker thread. We never do `future.await()` except in tests.

Please use `.then` to do that asynchronously. You can ping Core folks from storage team to have some guidance. cc @bbannier @chhsia0

Reply:
@GoelDeepak like this:
```cpp
return response
  .recover(defer(self(), [timeout](const Future<string>& future) {
    iptablesCommand = ...;
    return runScriptCommand(iptablesCommand, timeout);
  })
  .then([] { return Nothing(); });
```
The `recover` clause is called if `response` is transitioned into `Failed` or `Discarded`, which returns the result of another `runscriptCommand` retry. If you only want to retry when `response` fails, you can use `repair` instead.

So in the above snippet, if `response` is `Finished`, `response.recover(...)` will return the original future (i.e., `response`); if `response` is `Failed` or `Discarded`, `response.recover(...)` will return a new future (the retry one).

Reply:
Moreover, you can further do:
```cpp
return runScriptCommand(...)
  .recover(...)
  .then(...);
```